### PR TITLE
fix(go): add workaround to get semantic token highlighting

### PR DESCRIPTION
**Pull request by kcaashish**
Add a work around based on this comment: https://github.com/golang/go/issues/54531#issuecomment-1464982242
Related to #811

### CODE REVIEW

File: lua/lazyvim/plugins/extras/lang/go.lua [hunk 1]
```diff
@@ -14,13 +14,31 @@ return {
     "neovim/nvim-lspconfig",
     opts = {
       servers = {
-        gopls = {
-          settings = {
+        gopls = function(_, opts)
```
Comment (folke) [inline lua/lazyvim/plugins/extras/lang/go.lua:17]: This should always be a table. Did you mean to put it under `opts.setup`?

Comment (kcaashish) [reply lua/lazyvim/plugins/extras/lang/go.lua:17]: Oh, yes! It should be insider setup. Fixed that.

**Comment by folke** [comment]
thanks!

**Comment by fredrikaverpil** [comment]
I was hoping to enable inline SQL but it seems I have to set `semanticTokens = false` in the gopls settings to make that work. 😞 

```lua
  {
    "hexdigest/go-enhanced-treesitter.nvim",
    dependencies = {
      "derekstride/tree-sitter-sql",
    },
    build = ":TSInstall go sql",
    ft = "go",
  },
```

**Comment by hojjatabdollahi** [comment]
@fredrikaverpil did you ever figure out how to get the inline sql syntax highlighting to work? I tried to look at your dot files by they're too complex for my simple Go dev setup.

**Comment by fredrikaverpil** [comment]
@hojjatabdollahi yes, I got it working. You need to:

1. set gopls to not use semantic tokens: https://github.com/fredrikaverpil/dotfiles/blob/f55cfa07b53efed740b27ace830a1edf439bfc9f/nvim-fredrik/lua/fredrik/plugins/lang/go.lua#L221
2. add injections: https://github.com/fredrikaverpil/dotfiles/blob/f55cfa07b53efed740b27ace830a1edf439bfc9f/nvim-fredrik/after/queries/go/injections.scm#L10-L17

**Pull request merged**